src/txn/txn_ckpt.c

```diff
@@ -1263,8 +1263,13 @@ __checkpoint_lock_dirty_tree_int(WT_SESSION_IMPL *session, bool is_checkpoint, b
          * If there is a hot backup, don't delete any WiredTiger checkpoint that could possibly have
          * been created before the backup started. Fail if trying to delete any other named
          * checkpoint.
+         *
+         * N.B. Despite the comment above, this can cause corruption in the backup (WT-6118).
+         * For now we retain all WiredTiger checkpoints during a hot backup.  Once
+         * that is fixed the following condition should change to:
+         *     if (conn->hot_backup_start != 0 && ckpt->sec <= conn->hot_backup_start)
          */
-        if (conn->hot_backup_start != 0 && ckpt->sec <= conn->hot_backup_start) {
+        if (conn->hot_backup_start != 0) {
```

Comment:
A couple minor changes:
1. As a general rule, we don't reference or put ticket numbers into comments or code of the library. Sometimes in tests it is okay but we do try to avoid it if possible.
2. I think the undo change would be more obvious as an ifdef so that anyone who then fixes it later (likely you but possibly someone else) will obviously see the change and only has to remove the content in the ifdef without too much thought (or having to remember to read a comment above and change it). Something like:
```
#if 0
     if (conn->hot_backup_start != 0 && ckpt->sec <= conn->hot_backup_start) {
#else
    /*
     * Special case comment content here.
     */
    if (conn->hot_backup_start != 0) {
#endif